nagios/datadog_checks/nagios/nagios.py

```diff
@@ -331,14 +333,32 @@ def _parse_line(self, line):
     def create_event(self, timestamp, event_type, hostname, fields):
         """Factory method called by the parsers
         """
-        d = fields._asdict()
-        d.update({'timestamp': timestamp, 'event_type': event_type})
+        # Agent6 expects a specific set of fields, so we need to place all
+        # extra fields in the msg_title and let the Datadog backend separate them
+        # Any remaining fields that aren't a part of the datadog-agent payload
+        # specification will be dropped.
+        event_payload = fields._asdict()
+        for key, val in event_payload.iteritems():
+            event_payload[key] = val.lower()
+        msg_title = "event_soft_hard:" + event_payload.pop('event_soft_hard', " ")
+        msg_title += "|" + "event_type:" + event_payload.pop('event_type', " ")
+        msg_title += "|" + "check_name:" + event_payload.pop('check_name', " ")
+        msg_title += "|" + "event_state:" + event_payload.pop('event_state', " ")
+        msg_title += "|" + "payload:" + event_payload.pop('payload', " ")
+        msg_title += "|" + "ack_author:" + event_payload.pop('ack_author', " ")
+
+        event_payload.update({
+                'timestamp': timestamp,
+                'event_type': event_type,
+                'msg_title': msg_title,
+                'source_type_name': SOURCE_TYPE_NAME,
```

Comment:
I'm surprised you needed to add the `source_type_name` as well for Agent 6... Might be worth looking into a difference in behavior between Agent 5 and 6 there.

Have you noticed any discrepancy on the source type of events from other agent integrations? Typically they wouldn't show with the correct integration logo on the event stream

Reply:
The primary difference (of whether or not an Event submits with a source_type_name) should be that the event doesn't show up with the logo as you mentioned. 

Most other integrations emit this source_type_name field as part of the event that gets submitted. See [couchbase](https://github.com/DataDog/integrations-core/blob/master/couchbase/datadog_checks/couchbase/couchbase.py#L51) for example.

I think Agent 5 actually was looking for the source_type_name if it was defined and just defaulted to the name of the check otherwise - https://github.com/DataDog/dd-agent/blob/master/checks/collector.py#L443 (which is why it appears to work for A5 Nagios events and not for A6)

Reply:
I got bit with this a bit because I didn't realize Agent 5 was defaulting to the check name while Agent 6 submitted an empty value. We may want to have A6 send a default too.

Reply:
thanks for the investigation, at this point I don't think we should change the default behavior of Agent 6 on events: we may break other things without realizing it, and events are particularly finicky as you experienced :)

If we ever find other checks not setting the source type on events, and it has an impact on DD, let's fix the checks. Custom checks are unaffected since the list of source types are hardcoded in the backend anyway